### PR TITLE
Develop > Smart Contracts > Wasm (ink!)

### PR DESCRIPTION
**Pull request by 0xLucca**
This PR creates the "Wasm (ink!)" page

### CODE REVIEW

File: develop/smart-contracts/wasm-ink.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+- Comprehensive development tooling
+- Support from Rust's extensive developer community
+
+And because ink! smart contracts are compiled to Wasm, they deliver:
+
+- High execution speed
+- Platform independence
+- Enhanced security through sandboxed execution
+
+These contracts can be deployed on any blockchain built with the Polkadot SDK that implements the [Contracts Pallet](https://docs.rs/pallet-contracts/latest/pallet_contracts/){target=\_blank}.
```
Comment (nhussein11) [inline develop/smart-contracts/wasm-ink.md:25]: add a small sentence explaining the purpose of the `Contracts Pallet`

File: develop/smart-contracts/wasm-ink.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+## Installation
+
+ink! smart contract development requires the installation of [cargo-contract](https://github.com/use-ink/cargo-contract){target=\_blank}, a command-line interface (CLI) tool that provides essential utilities for creating, testing, and managing ink! projects.
+
+For step-by-step installation instructions, including platform-specific requirements and troubleshooting tips, refer to the official cargo-contract [Installation](https://github.com/use-ink/cargo-contract?tab=readme-ov-file#installation){target=\_blank} guide.
+
+## Quick Start
```
Comment (nhussein11) [inline develop/smart-contracts/wasm-ink.md:33]: This is very nitpicking, but I think that we are sticking to "Get Started" for this kind of titles

File: develop/smart-contracts/wasm-ink.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
+---
+title: Wasm (ink!)
+description: Learn to build Wasm smart contracts with ink!, a Rust-based eDSL. Explore installation, contract structure and key features.
```
Comment (nhussein11) [inline develop/smart-contracts/wasm-ink.md:3]:
```suggestion
description: Learn to build Wasm smart contracts with ink!, a Rust-based eDSL. Explore installation, contract structure, and key features.
```

File: develop/smart-contracts/wasm-ink.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+This command generates a new project directory with the following structure:
+
+```bash
+your-project-name/
+├── lib.rs          # Contract source code
+├── Cargo.toml      # Project configuration and dependencies
+└── .gitignore      # Git ignore rules
+```
+
+The `lib.rs` file includes a basic contract template with storage and message handling functionality. Customize this file to implement your contract’s logic. The `Cargo.toml` file defines project dependencies, including the necessary ink! libraries and configuration settings.
```
Comment (nhussein11) [inline develop/smart-contracts/wasm-ink.md:50]:
```suggestion
The `lib.rs` file includes a basic contract template with storage and message-handling functionality. Customize this file to implement your contract’s logic. The `Cargo.toml` file defines project dependencies, including the necessary ink! libraries and configuration settings.
```

File: develop/smart-contracts/wasm-ink.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+```
+
+!!!note
+    In this example, `new(init_value: u32)` initializes `number` with a specified value, while `default()` initializes it with the type’s default value (0 for `u32`). These constructors provide flexibility in contract deployment by supporting custom and default initialization options.
+
+For more information, refer to the official documentation for the [#[ink(constructor)]](https://use.ink/macros-attributes/constructor){target=\_blank} macro definition.
+
+### Messages
+
+Messages are functions used to interact with the contract, allowing users or other contracts to call specific methods. Each contract must define at least one message.
```
Comment (nhussein11) [inline develop/smart-contracts/wasm-ink.md:118]:
```suggestion
Messages are functions that interact with the contract, allowing users or other contracts to call specific methods. Each contract must define at least one message.
```

**Review by eshaben — APPROVED** [review]
LGTM!

**Pull request merged**